### PR TITLE
Implement features used by fenster

### PR DESCRIPTION
**Pull request by alanjian85**
[fenster-rv32emu](https://github.com/alanjian85/fenster-rv32emu), the rv32emu port of the Fenster library, utilizes capabilities not supported by rv32emu currently. These features include customizing window title, retrieving the mouse position and the program quit event. To support these features, the SDL-oriented system calls are refined to give the hosted program additional information and enable typical functions such as changing the title of the displayed window.

Note that these changes do not have binary backward compatibility, so all the binary files related to SDL-oriented system calls are updated and recompiled.

### CODE REVIEW

File: src/syscall_sdl.c [hunk 1]
```diff
@@ -338,6 +354,17 @@ void syscall_submit_queue(riscv_t *rv)
         case RELATIVE_MODE_SUBMISSION:
             SDL_SetRelativeMouseMode(submission.mouse.enabled);
             break;
+        case WINDOW_TITLE_SUBMISSION:
+            char *title = (char *) malloc(submission.title.size + 1);
```
Comment (jserv) [inline src/syscall_sdl.c]: No casting to `(char *)` is required.

**Review by jserv — CHANGES_REQUESTED** [review]
Update the content `docs/syscall.md` accordingly.

**Comment by jserv** [comment]
Thank @alanjian85 for contributing!

**Comment by alanjian85** [comment]
> Thank @alanjian85 for contributing!

By the way, the [fenster-rv32emu](https://github.com/alanjian85/fenster-rv32emu) repo is still under my personal account. It is better to fork and maintain it under this orgnanization.

**Pull request merged**